src/utils/getInteractiveRoles.ts

```diff
@@ -0,0 +1,18 @@
+import { ARIARoleDefinitionKey, roles } from "aria-query";
+
+const interactiveRoles: ARIARoleDefinitionKey[] = [];
+
+function getInteractiveRoles() {
+  for (const [role, definition] of roles.entries()) {
+    if (
+      !definition.abstract &&
+      definition.superClass.some((classes) => classes.includes("widget"))
+    ) {
+      interactiveRoles.push(role);
+    }
+  }
+
+  return interactiveRoles
+}
+
+export default getInteractiveRoles
```

Comment:
(important) This is kinda strange, if this method is ran multiple times, the `interactiveRoles` array will grow indefinitely.

Either define the const inside the function, or execute the function once and only export the return value of the function.

Reply:
Updated – moved const inside fn